eclair-node/src/main/scala/fr/acinq/eclair/Boot.scala

```diff
@@ -68,12 +72,21 @@ object Boot extends App with Logging {
     if (config.getBoolean("api.enabled")) {
       logger.info(s"json API enabled on port=${config.getInt("api.port")}")
       val apiPassword = config.getString("api.password") match {
-        case "" => throw EmptyAPIPasswordException
-        case valid => valid
+        case "" => None
+        case valid => Some(valid)
+      }
+      val apiCookiePassword = if (config.getBoolean("api.cookie-enabled")) {
+        Some(generateCookie(config.getString("api.cookie-path"), config.getBoolean("api.cookie-group-readable")))
+      } else {
+        None
+      }
+      if (apiPassword.isEmpty && apiCookiePassword.isEmpty) {
+        throw new RuntimeException("json-rpc api requires that either password or cookie is enabled")
       }
       val service: Service = new Service {
         override val actorSystem: ActorSystem = system
-        override val password: String = apiPassword
+        override val password: Option[String] = apiPassword
+        override val cookiePassword: Option[String] = apiCookiePassword
```

Comment:
We really shouldn't support both cookie and password auth at the same time.
The reason you would enable cookie authentication is to explicitly get rid of the user/password auth.
When cookie auth is enabled, this should automatically disable password auth.

I think we should force the user to clear its `password` field in `reference.conf`, like Bitcoin Core does, as this makes it very explicit for the user, and we should verify that they have done so.

Once that's done, I don't think we need to modify the `Service` class, there will still be a single `password` field that will be non-optional (containing either the user password or cookie password depending on which is enabled).